frappe/erpnext
# fix: Filter fields in E Commerce Settings (use Website Item)

Raised as improvised port of https://github.com/frappe/erpnext/pull/30179

**Issue:**
- While querying filter fields for `/all-products`, etc., the field filters were constructed by querying the **Item** master.
- This leads to scans over a larger table, half of which could be unpublished
- Also needs correctness

**Fix:**
- tweak `filters.py` to correctly query filter field values from Website Item (field filter contruction)
- Use **Website Item** for filter field options in **E Commerce Settings** and **Item Group**s  **Field Filter table**
- Use **Website Item** to validate field filters
- Make `validate_field_filters` reusable and reuse in Item Group as well

**UX:**
Changed **Filter Field** table to consist of **Autocomplete** fields in **E Commerce Settings** and **Item Group**
![2022-04-18 19 10 21](https://user-images.githubusercontent.com/25857446/163816546-2f2eb5f0-dccf-4ddf-8984-a33e0da0d185.gif)


**ToDo:**

- [x] test patch with table multiselect

**To Test: (pre merge)**
- On develop branch: Add a custom Link field to Item master and a Table Multiselect field
- Add these fields to Field filters in **E Commerce Settings**
- Switch to PR's branch and migrate site
- The new fields should be copied to website item (check)
- Check if product listing works fine with these
- Check if E commerce settings only allows fields from Website Item
